core/controllers/reader.py

```diff
@@ -281,12 +283,21 @@ def get(self, exploration_id):
                     'data_schema_version': data_schema_version
                 }
 
+        pretest_question_dicts = []
+        if story_id:
```

Comment:
I think you also need some verification that the exploration ID is in the given story, and raise 400 if not. Add backend test for this as well.

Reply:
Done.